relay-server/src/services/processor/span/processing.rs

```diff
@@ -413,6 +467,73 @@ fn validate(mut span: Annotated<Span>) -> Result<Annotated<Span>, anyhow::Error>
     if let Some(tags) = tags.value_mut() {
         tags.retain(|_, value| !value.value().is_empty())
     }
+    if let Some(measurements) = measurements.value_mut() {
+        measurements.retain(|_, value| match value.value() {
+            Some(v) => !v.is_empty() && !v.value.is_empty(),
```

Comment:
We might also want an `is_finite` check on `measurement.value`, because `serde_json` serializes infinites as `null`.